modules/clightning/module.cpp

```diff
@@ -235,7 +245,44 @@ std::string clightning_des_offer(const std::string_view input) {
         result << hex_encode(compressed, 33);
     }
     
-    clean_tmpctx();
+    return result.str();
+}
+
+std::optional<std::string> clightning_parse_p2p_lightning_message(std::span<const uint8_t> buffer) {
+    CleanTmpCtxGuard _cleanup;
+
+    u8 *msg = (u8 *) tal_arr(tmpctx, u8, buffer.size());
```

Comment:
I think `tal_arr` already does this cast:

```suggestion
    u8 *msg = tal_arr(tmpctx, u8, buffer.size());
```

Reply:
Updated!